addons/mail/static/src/messaging/component_hook/use_store/use_store.js

```diff
@@ -2,123 +2,17 @@ odoo.define('mail.messaging.component_hook.useStore', function (require) {
-         * impossible to compare them based on their current value (since it was
-         * updated in "both" due to the fact it is a proxy in the first place).
-         *
-         * And if the values are not proxy, `revNumber` will be 0 and the `isEqual`
-         * will be used to compare them.
-         */
-        let oldRevNumber;
-
-        function compare(a, b) {
-            let ok = true;
-            const newRevNumber = hashFn(b);
-            if (a === b && newRevNumber > 0) {
-                ok = oldRevNumber === newRevNumber;
-            } else {
-                ok = isEqual(a, b);
-            }
-            oldRevNumber = newRevNumber;
-            return ok;
-        }
-
-        return compare;
-    }
-
-    /**
-     * @see proxyComparator, but instead of comparing the given values, it compares
-     * their respective keys, with `compareDepth` level of depth.
-     * 0 = compare key, 1 = also compare subkeys, ...
-     *
-     * This assumes the given values are objects or arrays.
-     *
-     * @param {number|object} compareDepth the comparison depth, either as number
-     *  (applies to all keys) or as an object (depth for specific keys)
-     * @returns {function}
-     */
-    function proxyComparatorDeep(compareDepth = 0) {
-        const comparator = proxyComparator();
-        const comparators = {};
-
-        function compare(a, b) {
-            // If a and b are (the same) proxy, it is already managing the depth
-            // by itself, and a simple comparator can be used.
-            if (a === b && hashFn(b) > 0) {
-                return comparator(a, b);
-            }
-            let ok = true;
-            const newKeys = Object.keys(b);
-            if (!a || (Object.keys(a).length !== newKeys.length)) {
-                ok = false;
-            }
-            for (const key of newKeys) {
-                // the depth can be given either as a number (for all keys) or as
-                // an object (for each key)
-                let depth;
-                if (typeof compareDepth === 'number') {
-                    depth = compareDepth;
-                } else {
-                    depth = compareDepth[key] || 0;
-                }
-                if (!(key in comparators)) {
-                    if (depth > 0) {
-                        comparators[key] = proxyComparatorDeep(depth - 1);
-                    } else {
-                        comparators[key] = proxyComparator();
-                    }
-                }
-                // It is important to not break too early, the comparator has to
-                // be called for every key to remember their current states.
-                if (!comparators[key](a ? a[key] : a, b[key])) {
-                    ok = false;
-                }
-            }
-            return ok;
-        }
-
-        return compare;
-    }
-
+function useStore(selector, options) {
```

Comment:
I feel like changes to useStore should be reverted if possible, this does not seem necessary as the selectors return pretty much the same data as before.

Reply:
Hm, comparator assumes proxy has been proxified by OWL observer, which is not the case here with now just putting entity state in store and entity themselves are another kind of proxy.

I don't see any gain in maintaining this `useStore` at all, quite the contrary:
- it will be killed in the very near future
- need do to some effort to support 2 kinds of proxies in equality check (and certainly need to add meta-data on entity proxy to help this `useStore` handle efficient equality? definitely against that option!)

That's why I much prefer the overly generous re-render of default `useStore` for the time being.

Reply:
Until we get to that I think the selector should return the part of the entity that is in the store, not the entity itself then

Reply:
I'm ok for changes we'll do later, but only if we at least keep a status quo in the mean time. When there is a regression, it doesn't feel good.

Basically, if we had tests for "minimal rendering", we would have to skip them for this PR, which is never ideal if it can be avoided otherwise.

I need to finish the rest of the PR to decide, I have yet to see the technical part, I only see the "business" part up to that point.

Reply:
IMO it's a worse option than just using OWL `useStore`.

- If we use default `useStore`, we just have to remove this code (and we want to remove it, so this is a good step in the right direction).
- If we keep `useStore` implementation unchanged, we have to add code in all `useStore` that leak store state of entities... which goes against the intent to remove `useStore()` and its state management with mutable "observables".